modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/portal/widget/WidgetDispatcherApiHandler.java

```diff
@@ -0,0 +1,72 @@
+package com.enonic.xp.admin.impl.portal.widget;
+
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.google.common.collect.Multimap;
+
+import com.enonic.xp.web.WebException;
+import com.enonic.xp.web.WebRequest;
+import com.enonic.xp.web.WebResponse;
+import com.enonic.xp.web.universalapi.UniversalApiHandler;
+
+import static com.enonic.xp.admin.impl.portal.widget.GetListAllowedWidgetsHandler.WIDGET_INTERFACE_PARAM;
```

Comment:
Separate constants into own class. 
import static does not improve readability of code. It worsen especially in diffs. Please don't use them for anything but extremely common methods: `assertEquals`, `requireNonNull`